containers/README.md

```diff
@@ -0,0 +1,127 @@
+# GTSAM Containers
+
+- container files to build images
+- script to push images to a registry
+- instructions to pull images and run containers
+
+## Dependencies
+
+- a container engine such as [`Docker Engine`](https://docs.docker.com/engine/install/)
+
+## Pull from Docker Hub
+
+Various GTSAM image configurations are available at [`docker.io/borglab/gtsam`](https://hub.docker.com/r/borglab/gtsam). Determine which [tag](https://hub.docker.com/r/borglab/gtsam/tags) you want and pull the image.
+
+Example for pulling an image with GTSAM compiled with TBB and Python support on top of a base Ubuntu 22.04 image.
+
+```bash
+docker pull docker.io/borglab/gtsam:4.2.0-tbb-ON-python-ON_22.04
```

Comment:
Would it be possible to make the naming scheme: `gtsam:4.2.0-tbb-python_22.04`, `gtsam:4.2.0_22.04` etc?

Reply:
That would be pretty neat. I could change the name in the `hub_push.sh` scripts when it pushes to the image registry, however, I also specify the image:tag in the `compose.yaml` file. Here, we can do variable substitution, but I'm pretty sure we cannot do additional logic to have empty string instead of "tbb" based on the env variables.

I would like to keep both image names the same between the locally built image `compose.yaml` and the one that we push to docker hub `hub_push.sh`. And with that constraint I do not believe it is possible.

Another solution is to have a separate `compose.yaml` file for when TBB is ON and when python is ON, but then we could have a proliferation of `compose.yaml` files for each build configuration, and we wouldn't have a clean `hub_push.sh` script; it would be more complicated and have various nested if statements. I'd recommend against adding such complexity (even though it is 3 more compose.yaml files and a slightly more complicated bash script).